File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/model/KeyboardType.java
package space.lopatkin.spb.helpboardgamecard.domain.model;

public enum KeyboardType {
    CUSTOM, DEFAULT;

    public static KeyboardType getOrdinalFrom(String name) {
        for (KeyboardType type : values()) {
            if (type.name().equalsIgnoreCase(name)) {
                return type;
            }
        }
        return KeyboardType.CUSTOM;
    }

    public static KeyboardType getOrdinalFrom(int value) {
        for (KeyboardType type : values()) {
            if (type.ordinal() == value) {
                return type;
            }
        }
        return KeyboardType.CUSTOM;
    }

}

File: app/src/main/java/space/lopatkin/spb/helpboardgamecard/domain/model/KeyboardType.kt
package space.lopatkin.spb.helpboardgamecard.domain.model

enum class KeyboardType {
    CUSTOM,
    DEFAULT;

    companion object {

        fun getOrdinalFrom(name: String): KeyboardType {
            for (type in values()) {
                if (type.name.equals(other = name, ignoreCase = true)) {
                    return type
                }
            }
            return CUSTOM
        }

        fun getOrdinalFrom(value: Int): KeyboardType {
            for (type in values()) {
                if (type.ordinal == value) {
                    return type
                }
            }
            return CUSTOM
        }
    }

}
